doc/releases/release-notes-3.7.rst

```diff
@@ -353,6 +353,11 @@ Networking
 
   * Removed IPSP support. ``CONFIG_NET_L2_BT`` does not exist anymore.
 
+* TCP:
+
+  * ISN generation now uses SHA256 instead of MD5. Moreover it now relies on PSA APIs
```

Comment:
```suggestion
  * ISN generation now uses SHA-256 instead of MD5. Moreover it now relies on PSA APIs
```

SHA-256 is used elsewhere and this is often times added in the spec-definition of the algorithm (more searchable)